app/models/operatingsystems/redhat.rb

```diff
@@ -72,4 +76,21 @@ def pxe_kernel_options(params)
   def minor_version_help
     '0, 6.1810'
   end
+
+  def fedora?
```

Comment:
My biggest concern with these is that they're no longer a pure implementation of the `OperatingSystem` interface. So you can end up with a template that calls it on a Debian OS or something else. Perhaps make `fedora?` and `rhel_compatible?` private methods and implement `meets_requirement?` on `OperatingSystem`? For Debian/Ubuntu I can see similar uses.

Reply:
I have moved the `meets_requirement`  to the base OperatingSytem class, although each family would need to override it with the correct keyword arguments to make it more readable. In the current state, the code will not throw an error, but would return `false` for families that are not overriding the default implementation.
I do prefer this method of implementation, since it will make throw meaningful exceptions when incorrect arguments are used, unless you would like to see something like:
``` ruby
host.operatingsystem = ubuntu_20_04
do_something if host.operatingsystem.meets_requirement(rhel: 8)
```

As for `fedora?` and `rhel_compatible?`: I do think that these methods should be public, but on the derived class. So I would like to allow the following code:
``` ruby
do_something if host.operatingsystem.is_a(RedHat) && host.operatingsystem.fedora?
```